integration/sawtooth_integration/tests/intkey_client.py

```diff
@@ -31,9 +32,20 @@ def __init__(self, url, wait=30):
     def send_txns(self, txns):
         batch = self.factory.create_batch(txns)
 
-        response = self.send_batches(batch)
-        id_query = urlparse(response['link']).query
-        return parse_qs(id_query)['id'][0]
+        attempts = 25
```

Comment:
This should perhaps use exponential back-off up to some limit.  This would allow retrying sub-second initially but then back off to something slow like 8s or 16s.